internal/cloud/realm/client.go

```diff
@@ -91,5 +91,32 @@ func (c *client) do(method, path string, options api.RequestOptions) (*http.Resp
 		req.Header.Set(api.HeaderAuthorization, "Bearer "+auth)
 	}
 	client := &http.Client{}
-	return client.Do(req)
+	res, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode >= 200 && res.StatusCode <= 299 {
+		return res, nil
+	}
+
+	parsedErr := parseResponseError(res)
+	if err, ok := parsedErr.(ServerError); !ok {
```

Comment:
this is still missing the mark a bit, we need this:

```go
parsedErr := parseResponseError(res)
if err, ok := parsedErr.(ServerError); !ok {
  return nil, parsedErr
}
if options.PreventRefresh || err.Code != invalidSessionCode {
  return nil, err
}
```

Reply:
Yep, that was my bad from moving stuff around. Should fix the test failures too